### ISSUE TITLE
Clock example eats all memory after suspend/resume

### ISSUE THREAD

**Issue by shamrin** [witheve/lueve#3]
I was pleasantly surprised to find that counter and clock examples are finally working. Congratulations!

However, running clock example on current master (605e6ae948e72df1463d7d4cd281094ab2b583e8) gives me a bug with suspend/resume:
1. `git@github.com:witheve/lueve.git && cd lueve/build && make && cd ..`
2. `build/lueve examples/clock.eve`
3. Open Chrome at localhost:8080
4. Clock is working great!
5. Put laptop to sleep.
6. Wake up laptop after a few hours.

Result: Eve eats huge amounts of memory (10Gb+) and slows everything down while it's trying to keep up with current time. (I've terminated Eve process with Ctrl-C.)

**Comment by ibdknox** [comment]
Yeah, we turned off our reclamation logic right now so it's just eating all the memory forever lol. @convolvatron did some stuff this weekend to clean that up some, which we should be getting in today. Though another thing that I noticed myself this weekend is it appears somehow the timer continues to run during sleep. I wonder if the way that works isn't playing nicely with OSX somehow.

**Comment by shamrin** [comment]
> Though another thing that I noticed myself this weekend is it appears somehow the timer continues to run during sleep.

How did you notice it?

**Comment by ibdknox** [comment]
I only had the clock up for a few ticks when I closed the lid of my computer, when I looked again after waking it, there were thousands. That ended up being a red herring though, the culprit is that when the timer gets behind, it tries to make up all the old ticks. So when I woke my computer it just ran through all the old ones that it missed. While in some cases that might be the right behavior, I don't think it's what we want here for our default time implementation. I've created https://github.com/witheve/Eve/issues/369 to track that issue.

The memory leaks are getting a lot better, but there's still some more work to do.
